locked-asset/simple-lock-energy/src/unlock_with_penalty.rs

```diff
@@ -197,24 +203,92 @@ pub trait UnlockWithPenaltyModule:
+            if self.fees_from_penalty_unlocking().is_empty() {
+                // First fee deposit of the week
+                self.fees_from_penalty_unlocking()
+                    .set(NonceAmountPair::new(token_nonce, remaining_amount));
+            } else {
+                self.merge_fees_from_penalty(token_nonce, &remaining_amount)
+            }
+        }
+
+        // Send fees to FeeCollector SC
+        let current_epoch = self.blockchain().get_block_epoch();
+        let last_week_fee_sent_to_collector = self.last_week_fee_sent_to_collector().get();
+        if current_epoch >= last_week_fee_sent_to_collector + EPOCHS_PER_WEEK {
+            self.send_fees_to_collector();
+            self.last_week_fee_sent_to_collector().set(current_epoch);
         }
     }
 
-    fn send_fees_to_collector(&self, token_id: TokenIdentifier, amount: BigUint) {
+    /// Merges new fees with existing fees and saves in storage
+    fn merge_fees_from_penalty(&self, token_nonce: Nonce, new_fee_amount: &BigUint) {
+        let locked_token_mapper = self.locked_token();
+        let existing_nonce_amount_pair = self.fees_from_penalty_unlocking().get();
+        let existing_token_attributes: LockedTokenAttributes<Self::Api> =
+            locked_token_mapper.get_token_attributes(existing_nonce_amount_pair.nonce);
+
+        let mut output_pair = LockedAmountAttributesPair {
+            token_amount: existing_nonce_amount_pair.amount.clone(),
+            attributes: existing_token_attributes,
+        };
+
+        locked_token_mapper.nft_burn(
+            existing_nonce_amount_pair.nonce,
+            &existing_nonce_amount_pair.amount,
+        );
+
+        let new_token_attributes: LockedTokenAttributes<Self::Api> =
+            locked_token_mapper.get_token_attributes(token_nonce);
+        let new_pair = LockedAmountAttributesPair {
+            token_amount: new_fee_amount.clone(),
+            attributes: new_token_attributes,
+        };
+
+        locked_token_mapper.nft_burn(token_nonce, &new_fee_amount);
+
+        output_pair.merge_with(new_pair);
+
+        let sft_nonce = self.get_or_create_nonce_for_attributes(
+            &locked_token_mapper,
+            &output_pair.attributes.original_token_id.clone().into_name(),
+            &output_pair.attributes,
+        );
+
+        let new_locked_tokens =
+            locked_token_mapper.nft_add_quantity(sft_nonce, output_pair.token_amount.clone());
+
+        self.fees_from_penalty_unlocking().set(NonceAmountPair::new(
+            new_locked_tokens.token_nonce,
+            new_locked_tokens.amount,
+        ));
+    }
+
+    #[payable("*")]
+    #[endpoint(sendFeesToCollector)]
+    fn send_fees_to_collector(&self) {
         let sc_address = self.fees_collector_address().get();
+        let locked_token_id = self.locked_token().get_token_id();
+        let nonce_amount_pair = self.fees_from_penalty_unlocking().get();
```

Comment:
This can be called only ONCE per week.

Reply:
// Send fees to FeeCollector SC
        let current_epoch = self.blockchain().get_block_epoch();
        let last_week_fee_sent_to_collector = self.last_week_fee_sent_to_collector().get();
        if current_epoch >= last_week_fee_sent_to_collector + EPOCHS_PER_WEEK {
            Do all the code from here, plus the next line.
            self.last_week_fee_sent_to_collector().set(current_epoch);
        }

Reply:
Moved the "once per week" logic from `burn_penalty` inside `send_fees_to_collector`